lib/util/LocationStatusManager.js

```diff
@@ -0,0 +1,583 @@
+const async = require('async');
+const schedule = require('node-schedule');
+const { errors } = require('arsenal');
+const LocationStatus = require('../models/LocationStatus');
+
+const validLocations = require('../../conf/locationConfig.json') || {};
+const { locationStatusCollection } = require('../constants');
+
+const actions = {
+    deleteSchedule: 'deleteScheduledResumeService',
+    pause: 'pauseService',
+    resume: 'resumeService',
+};
+
+const supportedServices = ['lifecycle'];
+
+/**
+ * @class LocationStatusManager
+ *
+ * @classdesc manages the pause/resume status of the locations
+ * on different services.
+ */
+class LocationStatusManager {
+    constructor(mongoServices, mongoClient, zkClient, redis, topics, zkDetails, logger) {
+        this._mongoServices = mongoServices;
+        this._mongoClient = mongoClient;
+        this._zkClient = zkClient;
+        this._redis = redis;
+        this._topics = topics;
+        this._zkDetails = zkDetails;
+        this._logger = logger;
+
+        this._locationStatusColl = null;
+
+        this._locationStatusStore = {};
+        this._scheduledResumeJobs = {
```

Comment:
can this/should this be an attribute of `LocationStatus` ?
(and thus avaialble through the `locationStatusStore` map)

Reply:
`_scheduledResumeJobs` contains the actual scheduled jobs for all services (that use both mongo and zookeeper), this can't be put in the `LocationStatus` as it'll only work for mongo services